include/dl/dl.h

```diff
@@ -136,6 +136,8 @@ typedef enum
 	DL_TYPE_STORAGE_STR,
 	DL_TYPE_STORAGE_PTR,
 	DL_TYPE_STORAGE_STRUCT,
+	DL_TYPE_STORAGE_ANYPTR,
```

Comment:
shouldn't the storage here be just DL_TYPE_STORAGE_ANY and that combined with DL_TYPE_ATOM above would give you if it is an any_pointer, any_array or any_inline_array.

Reply:
Using storage prevented me from having anyarray[] which I really wanted without an extra struct in there.
The "any_inline_array" can't happen, since the "any" is of arbitrary size, so it needs a pointer (arrays use pointers) and thus can't be stored inline. If you want to store "any" inline then you must have a "size" and you can only store types smaller or equal to that size. Which basically makes it a union which you already support.

Reply:
You are ofcourse correct... I thought an inline array of anyptr... but that should just be "anyptr[16]" i guess

Reply:
Indeed, as shown with the type `WithInlineAnyptrArray` in the tests